### ISSUE TITLE
AutodetectMemoryLimitIT#testTooManyPartitions fails with too large a model

### ISSUE THREAD

**Issue by DaveCTurner** [elastic/elasticsearch#43013]
The following PR build failed: https://scans.gradle.com/s/yksgfi2gpxock/console-log?task=:x-pack:plugin:ml:qa:native-multi-node-tests:integTestRunner#L5 reporting the following:

```
org.elasticsearch.xpack.ml.integration.AutodetectMemoryLimitIT > testTooManyPartitions FAILED
java.lang.AssertionError:
Expected: a value less than <31500000L>
but: <31501528L> was greater than <31500000L>
at __randomizedtesting.SeedInfo.seed([97988765869B866C:A0C4E16B5AF81ECF]:0)
at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
at org.junit.Assert.assertThat(Assert.java:956)
at org.junit.Assert.assertThat(Assert.java:923)
at org.elasticsearch.xpack.ml.integration.AutodetectMemoryLimitIT.testTooManyPartitions(AutodetectMemoryLimitIT.java:82)
REPRODUCE WITH: ./gradlew :x-pack:plugin:ml:qa:native-multi-node-tests:integTestRunner --tests "org.elasticsearch.xpack.ml.integration.AutodetectMemoryLimitIT.testTooManyPartitions" -Dtests.seed=97988765869B866C -Dtests.security.manager=true -Dtests.locale=is -Dtests.timezone=SystemV/YST9 -Dcompiler.java=12 -Druntime.java=11
```

Looking at the build stats I see 17 failures of this test over the last 24 hours, all against `master`, but nothing prior to that.

**Comment by DaveCTurner** [comment]
Muted in 4810d3b5b20896d5740cf0a2940494779b07f461.

**Comment by droberts195** [comment]
This is most likely caused by elastic/ml-cpp#491.  Since that is due to be backported to 7.x I also backported the mute for this failure to 7.x in 68339f90e9b2ec84c70f4f064f9f0599791b265f.

**Comment by davidkyle** [comment]
Fix and unmuted in #43031. 
Closing.
